src/entities/friendship_history.rs

```diff
@@ -134,6 +142,45 @@ impl FriendshipHistoryRepository {
         }
     }
 
+    /// Fetches the request events of the given user.
+    pub async fn get_user_request_events(
```

Comment:
The name should reflect that only "pending" requests (sent or received) will be shown, something like `get_user_pending_requests`. Also, can we add tests for this query?